code/mergeblock/merge.py

```diff
@@ -56,31 +54,39 @@ def run():
     config = configparser.ConfigParser()
     config.read(config_file)
 
+    #open log file
+    logging.basicConfig(filename="mergeblock.log",level=logging.INFO)
+    logging.info("Program starts to run...")
+
     #read some settings from config file
     output_path = config['SETTINGS']['output']
     datalog_path = config['SETTINGS']['datalog']
     sleep_time = float(config['SETTINGS']['sleeptime'])
     line_length = int(config['SETTINGS']['line_length'])
     prep_text_packs = config['SETTINGS'].getboolean('text_pack')
-
     directories = get_directories(config)
     keys = get_keys(config)
 
     while True:
-        tails = get_tails(directories,line_length)
-        dict_pack = prepare_dict(tails,keys)
-        
-        if prep_text_packs:
-            datalog = open(datalog_path,"a")
-            datalog.write(str(dict_pack) + '\n')
-            datalog.close()
-
-        msg_packet = prepare_message_pack(dict_pack)
-        output = open(output_path,"ab")
-        output.write(msg_packet)
-        binary_new_line(output)
-        output.close()
-
-        time.sleep(sleep_time)
+        try:
+            tails = get_tails(directories,line_length)
+            dict_pack = prepare_dict(tails,keys)
+
+            if prep_text_packs:
+                datalog = open(datalog_path,"a")
+                datalog.write(str(dict_pack) + '\n')
+                datalog.close()
+
+            msg_packet = prepare_message_pack(dict_pack)
+            output = open(output_path,"ab")
+            output.write(msg_packet)
+            binary_new_line(output)
+            output.close()
+            time.sleep(sleep_time)
+
+        except Exception:
+            logging.exception("Unexpected Exception!")
+            time.sleep(0.1)
```

Comment:
Its no use to try immediately, also this protects flooding our log file with millions of exceptions

Reply:
i mean sleeping for 0.1 second